dateparser/search/search.py

```diff
@@ -171,15 +172,23 @@ def __init__(self):
         self.available_language_map = self.loader.get_locale_map()
         self.search = _ExactLanguageSearch(self.loader)
 
-    def detect_language(self, text, languages):
-        if isinstance(languages, (list, tuple, Set)):
+    @apply_settings
+    def detect_language(self, text, languages, settings=None, detect_languages_func=None):
+        if detect_languages_func:
+            detected_languages = detect_languages_func(text, settings=settings) or None
+            detected_languages = map_languages(detected_languages)
+            if not detected_languages:
+                if not settings.LANGUAGE_DETECTION_STRICT_USE and settings.DEFAULT_LANGUAGES:
+                    detected_languages = settings.DEFAULT_LANGUAGES[0]
+                return detected_languages
+            return detected_languages[0]
```

Comment:
why do we return only the first language?

Reply:
This function `detect_language` of `search_dates`  is required to return one language only.

Reply:
Ah I didn't know that, thanks!